shadows/framework/src/main/java/org/robolectric/shadows/ShadowBiometricManager.java

```diff
@@ -37,21 +42,50 @@ public class ShadowBiometricManager {
   protected int canAuthenticate() {
     if (RuntimeEnvironment.getApiLevel() >= R) {
       return reflector(BiometricManagerReflector.class, realBiometricManager).canAuthenticate();
-    } else if (biometricServiceConnected) {
-      return BIOMETRIC_SUCCESS;
     } else {
-      boolean hasBiomatrics =
-          ReflectionHelpers.callStaticMethod(
-              BiometricManager.class,
-              "hasBiometrics",
-              ClassParameter.from(
-                  Context.class, RuntimeEnvironment.getApplication().getApplicationContext()));
-      if (!hasBiomatrics) {
-        return BIOMETRIC_ERROR_NO_HARDWARE;
-      } else {
-        return BIOMETRIC_ERROR_HW_UNAVAILABLE;
-      }
```

Comment:
I removed this part because there is no test cases for it and i cannot find any "hasBiometrics" method in the original BiometricManager class

Reply:
Maybe `hasBiometrics` only existed in some old API source code, e.g. https://android.googlesource.com/platform/prebuilts/fullsdk/sources/android-29/+/refs/heads/androidx-recyclerview-recyclerview-selection-release/android/hardware/biometrics/BiometricManager.java#77. Maybe we should restore it, because there are many Google's internal tests maybe use this behavior.

Reply:
Oh didn't know that, lemme add it back. Thanks for sharing this, glad to know that backward compatibility is not an easy thing 🙏